challenges-api/src/main/java/org/exoplatform/challenges/service/ChallengeService.java

```diff
@@ -15,7 +16,7 @@
    * @throws IllegalAccessException when user is not authorized to create a
    *           challenge for the designated owner defined in object
    */
-  Challenge createChallenge(Challenge challenge, String username) throws IllegalAccessException;
+  ChallengeRestEntity createChallenge(Challenge challenge, String username) throws IllegalAccessException;
```

Comment:
We should not use the Rest entity here, let it be just on Rest services layer. In services layer we use DTOs that could be used to create Rest entities on upper layer.
We can have a new class Rest Entity Builder that will be used just for building Rest responses.